lib/preview/previewUtils.js

```diff
@@ -152,6 +154,20 @@ const getAuthType = (accountId) => {
   return authType;
 };
 
+const isUngatedForPreview = async (sessionInfo) => {
+  const { portalId, env, personalAccessKey } = sessionInfo;
+
+  const { enabledFeatures = {} } = await unAuth.fetchAccessToken(
```

Comment:
interesting -- so this will return an array of _all_ gates the portal is ungated to?

Reply:
All gates that are listed in this mapping as externally visible, very limited subset of gates for the portal: [see](https://git.hubteam.com/HubSpotProtected/LocalDevAuth/blob/393b2e0af81f69a0091e047cb1ffbfca3cd6335e/LocalDevAuthService/src/main/java/com/hubspot/localdevauth/service/LocalDevAuthServiceModule.java#L143)

Reply:
Okay so theres a filtered subset of external gates that are visible -- who/what determines if a gate is externally visible? Is this a system that we can just continually add gates to as we need them in the CLI?

Reply:
Yep! If you go through PR comments / code you'll see there's a little nuance (since any published version of the CLI can be running on a user's machine, gates shouldn't be removed from this enum, else a check could happen for a gate that's since been removed and crash things). But otherwise it's a relatively simple BE change.